rhessys/tec/input_new_patch.c

```diff
@@ -82,57 +84,46 @@
 	int		landuse_default_object_ID;
 	char		record[MAXSTR];
 	double	mpar, ltmp;
-
+	int		paramCnt=0;
+	param		*paramPtr=NULL;
 	
 	/*--------------------------------------------------------------*/
 	/*	Read in the next patch record for this hillslope.			*/
 	/*--------------------------------------------------------------*/
- 	fscanf(world_file,"%lf",&(ltmp));
-	read_record(world_file, record);
+	paramPtr = readtag_worldfile(&paramCnt,world_file,"Patch");
+
+	ltmp = getDoubleWorldfile(&paramCnt,&paramPtr,"x","%lf",patch[0].x,1);		
```

Comment:
@xiaoli-chen For input_new_X_*(), the default value of each variable should be the current value of the variable.  This way, redefine worldfiles only need to contain the variables to be redefined.